app/controllers/api/v2/users_controller.rb

```diff
@@ -0,0 +1,98 @@
+class Api::V2::UsersController < ApplicationController
+  include ApiHelper
+
+  def index
+    # TODO: ADD PERMISSIONS
+
+    res = User.find_by_sql "select * from users"
+
+    # HACK TO PREVENT AS_JSON OVERRIDE (DEF AS_JSON IN USER.RB)
+    # TODO: REMOVE ONCE DEF AS_JSON IS FIXED
+    result = []
+    res.each do |r|
+      result << { :id => r.id, :name => r.name, :login => r.login }
+    end
+
+    render json: api_ok(result) and return
+  end
+
+  def user
+    # TODO: ADD PERMISSIONS
+
+    # GET USER
+    id = params[:id].to_i
+    user = User.find(id) rescue nil
+    render json: api_error( { "user_id" => ["invalid user"] } ) and return if !user
+
+    # HACK TO PREVENT AS_JSON OVERRIDE (WHICH IS REALLY INEFFICENT BY THE WAY)
+    result = { :id => user.id, :name => user.name, :login => user.login }
+
+    render json: api_ok(result)
+  end
+
+  def jobs
+    # TODO: ADD PERMISSIONS
+
+    # GET USER
+    id = params[:id].to_i
+    user = User.find(id) rescue nil
+    render json: api_error( { "user_id" => ["invalid user"] } ) and return if !user
+
+    # HARDCODED JOB STATUSES
+    job_statuses = { "pending" => -1, "running" => 0, "done" => -2 }
+
+    pcs = [ ]
+    params[:status].to_a.each do |s|
+      pcs << job_statuses[s] if job_statuses[s]
+    end
+
+    if pcs == [ ]
+      render json: api_ok(user.jobs)
+    else
+      render json: api_ok(user.jobs.where("pc in (#{pcs.join(',')})"))
+    end
+  end
+
+  def assigned_jobs
+    # TODO: ADD PERMISSIONS
+
+    # GET USER
+    id = params[:id].to_i
+    user = User.find(id) rescue nil
```

Comment:
personally I seriously dislike the <model>.find(<id>) which bombs if the <id> does not exist, so I put the "rescue" in THAT call.

Then I used our standard practice of handling the exception first with a "render and return" 

I like what I did because it follows our norm.